src/keymap.cpp

```diff
@@ -776,7 +793,10 @@ std::vector<MacroCmd> KeyMap::parse_operation_sequence(const std::string& line)
 		cmds.push_back(cmd);
 	}
 
-	return cmds;
+	return ParsedOperations{
+		.operations = cmds,
+		.description = std::string(description)
+	};
```

Comment:
That's a nice one :) This shouldn't have compiled — this is [a designated initializer](https://en.cppreference.com/w/cpp/language/aggregate_initialization), which is only available in C++20 (and C99). I suppose it works because we don't use `-pedantic`, and since this feature is available in C99, even GCC 5.0 supports it.

No need to change this, just wanted to make this note so you're aware.

Reply:
I wasn't aware, thanks!